source/common/ssl/context_config_impl.cc

```diff
@@ -46,6 +46,7 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::CommonTlsContext& con
       private_key_file_(config.tls_certificates().empty()
                             ? ""
                             : config.tls_certificates()[0].private_key().filename()),
+      certificate_revocation_list_(config.validation_context().crl().filename()),
```

Comment:
The Datasource could either provide a filename or inline data.  Please handle both.  Instead of the ContextConfigImpl returning a path, it should just return the data.  If the source is a file, read it here in the constructor.  Then this class can probably hold a bssl::UniquePtr<X509_CRL> instead of std::string, and any invalid-data-errors will be caught sooner.

Reply:
Hum.  I'll admit that I used `Datasource` because `trusted_ca` was one, but it appears that field doesn't support non-file data either (see [here](https://github.com/envoyproxy/envoy/blob/master/source/common/ssl/context_config_impl.cc#L42)).  I'm happy to add support for inline data here, if you'd like, but do you think it's worth dropping support for inline data entirely?

Reply:
Yeah, I think the reason that many of the fields don't support inline data yet is that the boringssl API makes it really easy to just give it a file path and never touch the file (eg SSL_CTX_use_PrivateKey_file()).  In this case, since you're doing the file open manually, it seems less clear.  

But I think at some point, a separate cleanup pass will be needed to make everything support inline data.  If you want to defer fixing this until then, I'm ok with that. 

But if you do defer it, please rename the variables/functions to be clear that they hold a path/filename (eg private_key_file_/privateKeyFile()).

Reply:
I think @htuch wrote the original code that didn't handle inline data.  Any comment on if/when to handle this?

Reply:
Either way, I just pushed a commit that should parse the CRL up-front, and uses a memory BIO to handle inline data.  I'm not _super_ happy with it, since it involves exposing the `bssl::UniquePtr<X509_CRL>` in the public interface of `class ContextConfig`.

I'll write tests next!